### PR TITLE
add juniperServiceClient

### PR DESCRIPTION
**Pull request by ayrat-molotkov**

### CODE REVIEW

File: HappyTravel.JuniperConnector.Common/JuniperService/JuniperServiceClient.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        _staticDataClient = new StaticDataTransactionsClient(GetBasicHttpBinding("JuniperStaticDataServiceSoap"), GetEndpointAddress(_options.StaticDataEndPoint));
+        _staticDataClient.Endpoint.EndpointBehaviors.Add(new HttpMessageHandlerBehavior(factory, Constants.HttpStaticDataClientNames));
+        _availClient = new AvailTransactionsClient(GetBasicHttpBinding("JuniperAvailServiceSoap"), GetEndpointAddress(_options.AvailEndPoint));
+        _availClient.Endpoint.EndpointBehaviors.Add(new HttpMessageHandlerBehavior(factory, Constants.HttpAvailClientNames));
+        _login = GetLogin();
+    }
+
+
+    private BasicHttpBinding GetBasicHttpBinding(string name)
+            => new BasicHttpBinding(BasicHttpSecurityMode.Transport)
```
Comment (DFSko) [inline HappyTravel.JuniperConnector.Common/JuniperService/JuniperServiceClient.cs:23]: wrong indentation, should be 4 spaces

File: HappyTravel.JuniperConnector.Common/JuniperService/JuniperServiceClient.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                SendTimeout = TimeSpan.FromMinutes(5)
+            };
+
+
+    private EndpointAddress GetEndpointAddress(string endPoint)
+        => new EndpointAddress(new Uri(endPoint));
+
+
+    private JP_Login GetLogin()
+            => new JP_Login
```
Comment (DFSko) [inline HappyTravel.JuniperConnector.Common/JuniperService/JuniperServiceClient.cs:38]: wrong indentation, should be 4 spaces

**Review by kirillta — COMMENTED** [review]
Why do you use codegen here anyway?

File: HappyTravel.JuniperConnector.Common/JuniperService/JuniperServiceClient.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                MaxBufferSize = 20000000,
+                SendTimeout = TimeSpan.FromMinutes(5)
+            };
+
+
+    private EndpointAddress GetEndpointAddress(string endPoint)
+        => new EndpointAddress(new Uri(endPoint));
+
+
+    private JP_Login GetLogin()
```
[the remark this answers is not in the document]
Comment (ayrat-molotkov) [reply HappyTravel.JuniperConnector.Common/JuniperService/JuniperServiceClient.cs:37]: This model is generated automatically from soap

File: HappyTravel.JuniperConnector.Common/JuniperService/JuniperServiceClient.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+public class JuniperServiceClient : IJuniperServiceClient
+{
+    public JuniperServiceClient(IHttpMessageHandlerFactory factory, IOptions<ApiConnectionSettings> options)
+    {
+        _options = options.Value;
+        _staticDataClient = new StaticDataTransactionsClient(GetBasicHttpBinding("JuniperStaticDataServiceSoap"), GetEndpointAddress(_options.StaticDataEndPoint));
+        _staticDataClient.Endpoint.EndpointBehaviors.Add(new HttpMessageHandlerBehavior(factory, Constants.HttpStaticDataClientNames));
+        _availClient = new AvailTransactionsClient(GetBasicHttpBinding("JuniperAvailServiceSoap"), GetEndpointAddress(_options.AvailEndPoint));
+        _availClient.Endpoint.EndpointBehaviors.Add(new HttpMessageHandlerBehavior(factory, Constants.HttpAvailClientNames));
+        _login = GetLogin();
```
[the remark this answers is not in the document]
Comment (ayrat-molotkov) [reply HappyTravel.JuniperConnector.Common/JuniperService/JuniperServiceClient.cs:18]: This model will be needed for queries

File: HappyTravel.JuniperConnector.Common/JuniperService/JuniperServiceClient.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+using System.ServiceModel;
+
+namespace HappyTravel.JuniperConnector.Common.JuniperService;
+
+public class JuniperServiceClient : IJuniperServiceClient
+{
+    public JuniperServiceClient(IHttpMessageHandlerFactory factory, IOptions<ApiConnectionSettings> options)
+    {
+        _options = options.Value;
+        _staticDataClient = new StaticDataTransactionsClient(GetBasicHttpBinding("JuniperStaticDataServiceSoap"), GetEndpointAddress(_options.StaticDataEndPoint));
```
[the remark this answers is not in the document]
Comment (ayrat-molotkov) [reply HappyTravel.JuniperConnector.Common/JuniperService/JuniperServiceClient.cs:14]: Http clients is used in both api and updater. If register clients in startup, there will be code duplication

[the remark this answers is not in the document]
Comment (kirillta) [reply HappyTravel.JuniperConnector.Common/JuniperService/JuniperServiceClient.cs:14]: A constructor is not a place to execute so much code

File: HappyTravel.JuniperConnector.Common/HappyTravel.JuniperConnector.Common.csproj [hunk 6]
```diff
@@ -7,8 +7,14 @@
...
 
   <ItemGroup>
+
+    <PackageReference Include="Microsoft.Extensions.Http" Version="6.0.0" />
     <PackageReference Include="Microsoft.Extensions.Diagnostics.HealthChecks.EntityFrameworkCore" Version="6.0.5" />
     <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
+    <PackageReference Include="System.ServiceModel.Duplex" Version="4.8.*" />
+    <PackageReference Include="System.ServiceModel.Http" Version="4.8.*" />
+    <PackageReference Include="System.ServiceModel.NetTcp" Version="4.8.*" />
+    <PackageReference Include="System.ServiceModel.Security" Version="4.8.*" />
```
Comment (kirillta) [inline HappyTravel.JuniperConnector.Common/HappyTravel.JuniperConnector.Common.csproj:17]: Wildcard updates are dangerous

**Review by DFSko — APPROVED** [review]

**Pull request merged**